### PR TITLE
feat: adding info - is_success

### PR DESCRIPTION
**Pull request by GianiStatie**
Adding the ability to pass game info to the Python code, specifically if a run `is_success`. This gives more info to the developer when a simulation ends and they want to know the outcome.

This MR links to: 
* this Git issue: https://github.com/edbeeching/godot_rl_agents/issues/175

And is blocked by:
* this MR: https://github.com/edbeeching/godot_rl_agents_plugin/pull/46

### CODE REVIEW

**Comment by GianiStatie** [comment]
For now I'll update the code to set `info = None` as default, if the environments do not support it yet. 

I will open another issue to update the examples such that they return the correct value, once this and the linked MRs are merged.

**Review by Ivan-267 — COMMENTED** [review]
Hey, thanks for adding this feature! I haven't reviewed it in-depth yet, so this is just some initial feedback.

> I will open another issue to update the examples such that they return the correct value, once this and the linked MRs are merged.

I wonder if it might be better to keep this backwards compatible, such that it doesn't break existing environments, this makes it easier for people to upgrade (or use older plugin versions with the current GDRL). While not all updates are backward compatible, when it's easy to do it can be an advantage.

At some future point we can remove this backward compatibility. Since you already implemented the backward compatibility, I would personally recommend it to stay for now.

File: godot_rl/core/godot_env.py
Comment (Ivan-267) [inline godot_rl/core/godot_env.py:204]:
```suggestion
        # Kept for backward compatibility if the plugin doesn't send info.
```

**Review by Ivan-267 — APPROVED** [review]
I think this one is fine - but we merge this, I'd like to check the plugin and make some suggestions there.

**Comment by Ivan-267** [comment]
I'll push the docs to this PR as it's connected to the plugin, no need to start a new one.

**Review by Ivan-267 — APPROVED** [review]
This one should be ready - pending review by @edbeeching 

Again, thanks very much for implementing this.

**Comment by GianiStatie** [comment]
> Again, thanks very much for implementing this.

It wasn’t a big deal. Thanks for doing the tutorial and being so tedious 🙏

**Review by edbeeching — APPROVED** [review]
Thanks, LGTM

**Pull request merged**
